### PR TITLE
refactor(search-migration): migrate every search event with custom data 

### PR DESCRIPTION
**Pull request by alexprudhomme**
https://coveord.atlassian.net/browse/KIT-2912

The migration includes:

Marking the old log functions with `//TODO: KIT-2859`.
Adjusting every reference of every particular function to use the new 'next' analytics pattern.
Add tests for each deprecated functions to see if legacy analytics === next analytics in the search request.

Migrations: 

- https://github.com/coveo/analytics-migration-project/issues/13
- https://github.com/coveo/analytics-migration-project/issues/18
- https://github.com/coveo/analytics-migration-project/issues/14
- https://github.com/coveo/analytics-migration-project/issues/20 
- https://github.com/coveo/analytics-migration-project/issues/21
- https://github.com/coveo/analytics-migration-project/issues/25  
- https://github.com/coveo/analytics-migration-project/issues/26 
- https://github.com/coveo/analytics-migration-project/issues/29 
- https://github.com/coveo/analytics-migration-project/issues/30  
- https://github.com/coveo/analytics-migration-project/issues/31  
- https://github.com/coveo/analytics-migration-project/issues/43
- https://github.com/coveo/analytics-migration-project/issues/10 
- https://github.com/coveo/analytics-migration-project/issues/5
- https://github.com/coveo/analytics-migration-project/issues/6
- https://github.com/coveo/analytics-migration-project/issues/7

**_After these, [only edge cases are left](https://github.com/orgs/coveo/projects/10/views/6) (events that are not sent via `executeSearch`)_**

### CODE REVIEW

File: packages/headless/src/integration-tests/analytics-migration.test.ts [hunk 1]
```diff
@@ -258,6 +368,281 @@ describe('Analytics Search Migration', () => {
...
+          ),
+      },
+    });
+
+    legacySearchEngine.dispatch(action);
+    nextSearchEngine.dispatch(action);
+    await wait();
+
+    assertNextEqualsLegacy(callSpy);
+  });
```
Comment (alexprudhomme) [inline packages/headless/src/integration-tests/analytics-migration.test.ts:414]: Weird case where the state absolutely had to have facets for the analytics to work. I added them manually with actions.

**Review by louis-bompart — COMMENTED** [review]
LGTM, great job :D 
However, could you UT the new methods in `packages/headless/src/api/analytics/search-analytics.ts` ?
